### DISCUSSION TITLE
Destroying store yet adhering to the "only export slices/custom hooks" rule

### DISCUSSION THREAD

**Discussion by viveleroi** [pmndrs/zustand#2025] [Q&A]
TkDodo and others [recommend only exporting custom hooks/slices](https://tkdodo.eu/blog/working-with-zustand).

I have my first store with subscribe listeners so instead of calling a custom `reset` method and `unsubscribe` functions I was thinking of using the `.destroy` method on the store itself.

However, if it's generally bad practice to do `const { destroy } = useMyStore()` then how should I calling the destroy method?

Per the recommendations we're not exporting the store, and only slices:

```ts
const useParameterGraphsStore = create(
  subscribeWithSelector(
    immer<ParameterGraphsState>(set => ({
      ...initialState,
      actions: {
        //...
      }
    }))
  )
)

export const useParameterGraphs = (): ParameterGraphSeries[] => useParameterGraphsStore(state => state.graphs)

export const useParameterGraphActions = (): ParameterGraphActions => useParameterGraphsStore(state => state.actions)
```

**Comment by dai-shi** [comment]
How about this?
```js
export const destroyParameterGraphs = useParameterGraphsStore.destroy
```

Note that, however, `.destroy` is deprecated and will be removed in the next major version.
